File: ReactNativeApp/android/app/src/main/java/com/components/basic/calibration/SensorCalibrationBuilder.java
package com.components.basic.calibration;

import com.ssvc.SSVC;
import com.ssvc.SSVCBuilder;

public class SensorCalibrationBuilder extends SSVCBuilder {
    static private SensorCalibrationState buildState() {
        return new SensorCalibrationState();
    }

    public static SSVC<SensorCalibrationState, SensorCalibrationView, SensorCalibrationController> buildSensorCalibration() {
        SensorCalibrationState state = new SensorCalibrationState();
        SensorCalibrationView sensorCalibrationView = new SensorCalibrationView(state);
        SensorCalibrationController sensorCalibrationController = new SensorCalibrationController(state);
        return new SSVC<>(state, sensorCalibrationView, sensorCalibrationController);
    }
}

File: ReactNativeApp/ios/Modules/SSVC/Instance/Calibration/Device/SensorCalibrationBuilder.swift
//
//  SensorCalibrationBuilder.swift
//  jogo
//
//  Created by Mohsin on 20/10/2021.
//

import Foundation

public class SensorCalibrationBuilder: SSVCBuilder {
  
  private static func buildState() -> SensorCalibrationState {
    return SensorCalibrationState()
  }
  
  public static func build() -> SSVC<SensorCalibrationState, SensorCalibrationView, SensorCalibrationController> {
    let state = buildState()
    let view = SensorCalibrationView(state: state)
    let controller = SensorCalibrationController(state: state)
    return SSVC(state: state, stateView: view, controller: controller)
  }
}
